flight/PiOS/posix/pios_thread.c

```diff
@@ -34,9 +34,57 @@ struct pios_thread
 {
 	pthread_t thread;
 
-	char *name;
+	const char *name;
 };
 
+/**
+ * @brief   Creates a handle for the current thread.
+ *
+ * @param[in] namep        pointer to thread name
+ *
+ * @returns instance of @p struct pios_thread or NULL on failure
+ */
+struct pios_thread *PIOS_Thread_WrapCurrentThread(const char *namep)
+{
+	struct pios_thread *thread = PIOS_malloc_no_dma(sizeof(struct pios_thread));
+
+	extern bool are_realtime;
+
+	if (!thread) {
+		abort();
+	}
+
+	thread->thread = pthread_self();
+
+	thread->name = namep;
+
+	if (are_realtime) {
+		struct sched_param param = {
+			.sched_priority = 30 + PIOS_THREAD_PRIO_HIGHEST * 5
+		};
+
+		pthread_setschedparam(thread->thread, SCHED_RR, &param);
+	}
+
+#ifdef __linux__
+	pthread_setname_np(thread->thread, thread->name);
```

Comment:
Nice!. Can we do this for the other threads too? (unless I missed something, I don't think we do)

Reply:
We do, actually :D

Reply:
Great. :D